core/types/uuid.go

```diff
@@ -43,21 +44,47 @@ func (u UUID) String() string {
 }
 
 func (u UUID) Value() (driver.Value, error) {
-	return u[:], nil // []byte works for sql
+	return u.Bytes(), nil
 }
 
-func (u UUID) Bytes() []byte {
+func (u *UUID) Bytes() []byte {
 	return u[:]
 }
 
-var _ driver.Valuer = UUID{}
+// Over json, we want to send uuids as strings
+func (u *UUID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(u.String())
```

Comment:
Some trouble here now.  You get different results if you marshal a value vs a pointer.   Here's where a value receiver for the marshal method is normally the way:

```go

func Test_UUIDJSON(t *testing.T) {
	seed := []byte("test")

	uuidPtr := types.NewUUIDV5(seed)
	uuidVal := *uuidPtr

	b, err := json.Marshal(uuidPtr)
	if err != nil {
		t.Fatal(err)
	}

	t.Log(string(b)) // "24aa70cf-0e18-57c9-b449-da8c9db37821"

	var uuid2 types.UUID
	err = json.Unmarshal(b, &uuid2)
	if err != nil {
		t.Fatal(err)
	}
	t.Log(uuid2) // 24aa70cf-0e18-57c9-b449-da8c9db37821

	b, err = json.Marshal(uuidVal)
	if err != nil {
		t.Fatal(err)
	}

	t.Log(string(b)) // [36,170,112,207,14,24,87,201,180,73,218,140,157,179,120,33] (oops, didn't use MarshalJSON method!)

	var uuid3 types.UUID
	err = json.Unmarshal(b, &uuid3)
	if err != nil {
		t.Fatal(err)
	}
	t.Log(uuid3) // json: cannot unmarshal array into Go value of type string
}
```